conf/battle/battle.conf

```
@@ -130,10 +130,19 @@ equip_self_break_rate: 100
 // This affects the behaviour of skills like acid terror and meltdown
 equip_skill_break_rate: 100
 
-// Do weapon attacks have a attack speed delay before actual damage is applied? (Note 1)
-// NOTE: The official setting is yes, even thought it degrades performance a bit.
+// Should damage have a delay before it is applied? (Note 1)
+// Some skills might not have a delay by default regardless of this setting.
+// The official setting is yes, even thought it degrades performance a bit.
 delay_battle_damage: yes
 
+// Should the damage timing be synchronized between the client and server? (Note 1)
+// This is not official behavior, but it removes the position lag after being hit.
+// This setting only affects normal monster attacks and takes priority over "delay_battle_damage".
+// Many skills show their damage immediately, so setting "delay_battle_damage" to "no" at the same
+// time might improve the experience further, but will not work for all skills.
```

```suggestion
// Many skills show their damage immediately, so setting "delay_battle_damage" to "no" at the same
// time might improve the experience further, but will not work for all skills.
```
Could we not add a check for that as well? Maybe in another PR?
Basically we would just have to replace the skill_id == 0 check with a skill_id == 0 || skill_get_delay == 0 (pseudo) check?

---

The correct way to implement this for skills is to define the clientamotion for each skill and then use this information when sychnronize_damage is enabled. Would be great, but currently we don't have this information available.